sdk/dapp-kit/src/hooks/wallet/useSignAndExecuteTransactionBlock.ts

```diff
@@ -84,32 +93,65 @@ export function useSignAndExecuteTransactionBlock({
 
 				return walletFeature.signAndExecuteTransactionBlock({
 					...signTransactionBlockArgs,
+					transactionBlock: await transactionBlock.toJSON({
```

Comment:
One implication of this is that for wallets like zkSend, this will end up being a wait before this method is called, which is a bit scary because the window might need to be opened before calling `toJSON`. Not sure a good way to work around that.